CHANGELOG.md

```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The default value for `draw_bold_text_with_bright_colors` is now `false`
 - Mirror OSC query terminators instead of always using BEL
 - Increased Beam, Underline, and Hollow Block cursors' line widths
+- Dynamic title is not disabled anymore when `window.title` is set in config
```

Comment:
I know this might be out of scope for this PR, but I find it very strange that the setting `window.title` and `dynamic_title` aren't both in the `window` group. Since we're changing the semantics now, maybe it's also a good time to move this configuration?

Otherwise I can open a separate PR, if we agree that this should be done.

Reply:
I think @kchibisov might want to play around with this for the configuration changes required for the GUI separation?

Reply:
That's fine, but we agree these two configurations should be together correct?

Reply:
It depends a bit on the future of our GUI/terminal separation.

Because dynamic_title is an inherent terminal emulation feature, while the current title itself does not need to be known at all by the terminal emulator if we decide to never support the title query escape.

Reply:
I've already proposed to group `dynamic_title` and `title` somehow(let's say `enum`), I'm not afraid of `yaml` enums here.

Reply:
I am, I still don't think enum is a great choice here.

Reply:
@kchibisov what is a YAML enum? I've never heard of this, and can't find it in the [YAML spec](https://yaml.org/spec/1.2/spec.html).

Reply:
@nixpulvis There is none. That's the problem.

A Rust enum in yaml looks like this:

```yml
enum Test {
    One,
    Two,
}

value: {
    Test: One,
}
```